src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -386,24 +387,26 @@ void GcsResourceManager::GetResourceUsageBatchForBroadcast(
     rpc::ResourceUsageBroadcastData &buffer) {
   absl::MutexLock guard(&resource_buffer_mutex_);
   resources_buffer_proto_.Swap(&buffer);
-  while (!resources_buffer_.empty() &&
-         buffer.ByteSizeLong() <
-             RayConfig::instance().resource_broadcast_batch_size_bytes()) {
-    auto element = std::begin(resources_buffer_);
-    buffer.add_batch()->mutable_data()->Swap(&element->second);
-    resources_buffer_.erase(element);
+  auto beg = resources_buffer_.begin();
```

Comment:
This is a little unfortunate that we rely on cnt here... @wuisawesome is there any good way to get some heuristic size of each batch without calling that method?

Reply:
I prefer the size much more than getting the actual byte-size. It's simpler, and solves the same problem. The threshold can always be lowered if there is any remaining issues.

Reply:
It actually can be done in an estimated way, for example, get the first several ones and then do extrapolation.
I thought about that, but later I think maybe it's not a good idea. 1) It's a little bit hard to understand how the num of RPC requests are sent. For example, with X nodes, we don't have a good estimation of how many RPCs we need for pull-push. Another one is the SerDe is too expensive for these messages, and we need to just bypass them in some way. For example, right now, the SerDe is O(N^2) [N=# of nodes], but actually, we can make it O(N), and we can cap later after optimization.